### ISSUE TITLE
Add a proper line/strip/fan meshes

### ISSUE THREAD

**Issue by louen** [STORM-IRIT/Radium-Engine#74]
The Renderer's Mesh now has a triangle mesh inside which make lines meshes very awkward ( read : extremely hacky). We need to handle them correctly.

**Comment by Zouch** [comment]
Not sure I understand what you mean there

**Comment by louen** [comment]
It's hard to draw lines, because the current implementation of Engine::Mesh has a triangle mesh. You can fit a line mesh in it but it's awkward.

**Comment by Zouch** [comment]
Ok I see what you mean and actually the problem is the same for points.
Maybe it could be worth templating the mesh with its positions / indices
data.

Do you see anything wrong with that ?

**Comment by louen** [comment]
Actually templating Core::TriangleMesh seems a good idea.

Something like 

```c++
template< typename Face>
class MeshBase { 
   Vector3Array vertex;
   Vector3Array normals;
   VectorArray<Face> faces;
};

typedef MeshBase<Triangle> TriangleMesh;
```

**Comment by Zouch** [comment]
Vendu !

**Comment by louen** [comment]
This is still relevant (see this function)
`void Mesh::loadGeometry( const Core::Vector3Array& vertices, const std::vector<uint>& indices )`

**Comment by dlyr** [comment]
Fixed by #454 
Even if there is still todos to have full coverage of any mesh type.

**Comment by nmellado** [comment]
Keeping this open, according to https://github.com/STORM-IRIT/Radium-Engine/issues/74#issuecomment-567806871

**Comment by nmellado** [comment]
@dlyr , what do you think ?
Do we still need this ? I mean, you improved the Mesh class, lines are well supported right ?

**Comment by hoshiryu** [comment]
It seems that some of the `RA_DISPLAY_` macros don't compile

**Comment by dlyr** [comment]
Well line mesh are ok, but strip and fan are still not supported in core mesh.
@hoshiryu could you point me which ?
I have conitnued a little bit to convert draw primitive creation ... so either we can close the issue or keep it as a reminder, but in any case it's in the road map.

**Comment by dlyr** [comment]
Got it for the macro, i'll fix that quickly

**Comment by nmellado** [comment]
@dlyr i updated the issue title so we can keep it open as reminder.

**Comment by dlyr** [comment]
Will be fix along with issue #603
